lib/matplotlib/patches.py

```diff
@@ -1037,13 +1037,24 @@ def _update_path(self):
                              f"`len(values) = {self._values.size}` and "
                              f"`len(edges) = {self._edges.size}`.")
         verts, codes = [], []
-        for idx0, idx1 in cbook.contiguous_regions(~np.isnan(self._values)):
+
+        _nan_vals = ~np.isnan(self._values)
+        if self._baseline is not None:
+            _nan_vals = _nan_vals & ~np.isnan(self._baseline)
```

Comment:
We want logical not bitwise operations here:
```suggestion
            _nan_vals = np.logical_and(_nan_vals, ~np.isnan(self._baseline))
```

Actually, it's simpler to invert the logic:

~~~
        nan_vals = np.isnan(self._values)
        if self._baseline is not None:
            nan_vals = np.logical_or(_nan_vals, np.isnan(self._baseline))
        for idx0, idx1 in cbook.contiguous_regions(~nan_vals):
~~~

Reply:
Those are the same for boolean arrays.

Reply:
Ok, so 
~~~
nan_vals |= np.isnan(self._baseline)
~~~